itou/siaes/management/commands/import_siae.py

```diff
@@ -275,8 +275,10 @@ def create_new_siaes(self):
 
     @timeit
     def check_whether_signup_is_possible_for_all_siaes(self):
-        for siae in Siae.objects.prefetch_related("memberships").filter(members__isnull=True):
-            if not siae.has_members and not siae.auth_email:
+        for siae in Siae.objects.prefetch_related("members", "members__user").filter(
+            members__isnull=True, members__user__is_active=False
```

Comment:
Pour ma curiosité quel est l'intérêt d'avoir rajouté  la clause `members__user__is_active=False` ? La clause `members__isnull=True` garantissant déjà qu'on aura que des siaes avec aucun `members` non ?

En fait j'ai même un doute sur l'interprétation de ce filter. Est-ce qu'il signifie qu'on prend les siaes sans membre, ou bien les siaes sans membre inactif (mais potentiellement avec des membres actifs) ?

Pour pas se prendre la tête on pourrait juste se contenter d'un `for siae in Siae.objects.prefetch_related("members", "members__user")` sans filtre, si le prefetch related est bon ce sera très rapide de toute façon et plus simple à lire.

Reply:
tu as raison le second filtre ne sert à rien, je suis allé un peu vite !

Reply:
Je corrige et on merge alors ?

Reply:
Yep 🙏  @vperron

Reply:
En fait non, ça casse mon test de faire ça. Du coup je pense que je vais le laisser mais que penses-tu de ma logique:
- `members__isnull=True` retire les Siae sans membres
- ET `members__user__is_active=False` retire les SIAE dont TOUS les membres sont inactifs (quasi certain mais tu me mets le doute...)

... Du coup je me dis que le `if not siae.has_members` qui arrive derrière est inutile, puisque on a déjà filtré là dessus; mais il sert de garde-fou si je me trompe sur le filtre. Faudrait que je teste mes dires en créant un test avec une SIAE qui des users actifs ET inactifs et voir si elle est sélectionnée.

Reply:
Du coup y aurait encore un peu de travail.

Reply:
Ah la fameuse subtilité `.filter(A).filter(B)` vs `filter(A, B)`...

Mes 2 cts : simplifions-nous la vie, vire le filtre complètement. Ca serait un chouilla plus long (et encore, avec ton prefetch ça devrait prendre ~1sec) mais tellement plus lisible.

```
for siae in Siae.objects.prefetch_related("members", "members__user").all():
    if not siae.auth_email and not siae.has_members:
```

Reply:
Sans le premier filtre ni le premier je fais 4 queries.
Sans le second filtre j'en fais 2.
Avec les deux j'en fais une. Du coup ?

Reply:
J'ai assez envie du coup de laisser tous les filtres puisque en réalité c'est ça qui accélère le plus, de retirer le IF derrière et de faire un test par contre avec une SIAE qui a:
- un membre actif
- un membre inactif
et vérifier aussi le retour de la fonction. Ca te va ?

Reply:
Ok noice @vperron

Reply:
Je suis encore dessus, c'est bien plus complexe (et intéressant) que prévu, mais j'ai plein d'autres tickets plus urgents. Je te tiendrai au courant quand j'aurai le fin mot mais ça sent l'analyse fun !